stdlib/_interpchannels.pyi

```diff
@@ -0,0 +1,86 @@
+from _typeshed import structseq
+from typing import Any, Final, Literal, TypeVar, final
+from typing_extensions import Buffer, Self, TypeAlias
+
+_T = TypeVar("_T")
+_SendType: TypeAlias = Literal["send", "recv", "both"]
+
+class ChannelError(RuntimeError): ...
+class ChannelClosedError(ChannelError): ...
+class ChannelEmptyError(ChannelError): ...
+class ChannelNotEmptyError(ChannelError): ...
+class ChannelNotFoundError(ChannelError): ...
+
+@final
+class ChannelID:
+    @property
+    def end(self) -> _SendType: ...
+    @property
+    def send(self) -> Self: ...
+    @property
+    def recv(self) -> Self: ...
+    def __eq__(self, other: object) -> bool: ...
+    def __ge__(self, other: object) -> bool: ...
+    def __gt__(self, other: object) -> bool: ...
+    def __hash__(self) -> int: ...
+    def __index__(self) -> int: ...
+    def __int__(self) -> int: ...
+    def __le__(self, other: object) -> bool: ...
+    def __lt__(self, other: object) -> bool: ...
+    def __ne__(self, other: object) -> bool: ...
+
+@final
+class ChannelInfo(structseq[Any], tuple[bool, bool, bool, int, int, int, int, int]):
+    __match_args__: Final = (
+        "open",
+        "closing",
+        "closed",
+        "count",
+        "num_interp_send",
+        "num_interp_send_released",
+        "num_interp_recv",
+        "num_interp_recv_released",
+    )
+    @property
+    def open(self) -> bool: ...
+    @property
+    def closing(self) -> bool: ...
+    @property
+    def closed(self) -> bool: ...
+    @property
+    def count(self) -> int: ...  # type: ignore[override]
+    @property
+    def num_interp_send(self) -> int: ...
+    @property
+    def num_interp_send_released(self) -> int: ...
+    @property
+    def num_interp_recv(self) -> int: ...
+    @property
+    def num_interp_recv_released(self) -> int: ...
+    @property
+    def num_interp_both(self) -> int: ...
+    @property
+    def num_interp_both_recv_released(self) -> int: ...
+    @property
+    def num_interp_both_send_released(self) -> int: ...
+    @property
+    def num_interp_both_released(self) -> int: ...
+    @property
+    def recv_associated(self) -> bool: ...
+    @property
+    def recv_released(self) -> bool: ...
+    @property
+    def send_associated(self) -> bool: ...
+    @property
+    def send_released(self) -> bool: ...
+
+def create() -> ChannelID: ...
+def destroy(cid: int) -> None: ...
```

Comment:
I think most of the functions that accept `cid` arguments are definitely able to accept `ChannelID` instances as well as `int` instances. They're possibly able to accept anything that has an `__index__` method (in which case we should use `typing.SupportsIndex`), but I haven't got time to dig through the C code to check that right now

Reply:
Let me take another look at the C code to confirm.

Reply:
Yeah, it seems so, also tested this out myself: https://github.com/python/typeshed/pull/12199/commits/d45c6f72e0b4effe9d84b07f59ef27b4c39d9966